### PR TITLE
*: balance based on read flow

### PR DESCRIPTION
**Pull request by UncP**

### CODE REVIEW

File: pdctl/command/hot_command.go [hunk 1]
```diff
@@ -21,8 +21,9 @@ import (
 )
 
 const (
-	hotRegionsPrefix = "pd/api/v1/hotspot/regions"
-	hotStoresPrefix  = "pd/api/v1/hotspot/stores"
+	hotRegionsPrefix     = "pd/api/v1/hotspot/regions"
+	hotReadRegionsPrefix = "pd/api/v1/hotspot/readregions"
```
Comment (disksing) [inline pdctl/command/hot_command.go]: Maybe `regions/write` and `regions/read`.

File: server/balancer.go [hunk 2]
```diff
@@ -791,3 +791,351 @@ func (h *balanceHotRegionScheduler) GetStatus() *StoreHotRegionInfos {
...
+			continue
+		}
+		if r.HotDegree < hotRegionLowThreshold {
+			continue
+		}
+
+		regionInfo := cluster.getRegion(r.RegionID)
+		leaderStoreID := regionInfo.Leader.GetStoreId()
+		storeIDs := regionInfo.GetStoreIds()
+		for storeID := range storeIDs {
```
Comment (disksing) [inline server/balancer.go]: No need the loop. We can update `leaderStoreID` directly.

File: pdctl/command/hot_command.go (hunk 1 above)
Comment (UncP) [reply pdctl/command/hot_command.go]: > "pd/api/v1/hotspot/regions"

this is for write, maybe
> "pd/api/v1/hotspot/read"

this is for read ?

**Comment by nolouch** [comment]
PTAL @disksing @zhangjinpeng1987 @siddontang

**Comment by disksing** [comment]
LGTM. PTAL @zhangjinpeng1987 @overvenus

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**